libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5348,6 +5348,22 @@ void GCS_MAVLINK::send_sys_status()
         errors4); // errors4
 }
 
+MAV_LANDED_STATE GCS_MAVLINK::landed_state() const
+{
+    switch (AP::vehicle()->get_landed_state()) {
+        case AP_Vehicle::LandedState::OnGround:
+            return MAV_LANDED_STATE_ON_GROUND;
+        case AP_Vehicle::LandedState::InAir:
+            return MAV_LANDED_STATE_IN_AIR;
+        case AP_Vehicle::LandedState::TakeOff:
+            return MAV_LANDED_STATE_TAKEOFF;
+        case AP_Vehicle::LandedState::Landing:
+            return MAV_LANDED_STATE_LANDING;
+        default:
+            return MAV_LANDED_STATE_UNDEFINED;
```

Comment:
Removed this.  If a new landed state is added the person who adds it must decide on an action here.

Of course you could do this function with a cast if you decide to go down the 1:1-guaranteed-via-assertion path.